### DISCUSSION TITLE
메모리에 저장된 값에 직접 접근하는 게 아닌 변수를 사용하는 이유가 무엇인가요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#14] [변수]
- 메모리에 저장된 값에 직접 접근하는 게 아닌 변수를 사용하는 이유가 무엇인가요?

**Comment by pipisebastian** [comment]
>메모리에 저장된 값에 직접 접근하는 게 아닌 변수를 사용하는 이유가 무엇인가요? 

실제 물리 메모리 주소는 운영체제가 관리를 합니다. 실제로 운영체제가 관리중인 메모리를 변경하게 될 경우, 실행중인 프로그램에 치명적인 버그가 생길 수 있습니다. 
이에 변수를 두어, 메모리 주소에 직접 접근하는 것을 막습니다.

**Comment by publdaze** [comment]
> - 메모리에 저장된 값에 직접 접근하는 게 아닌 변수를 사용하는 이유가 무엇인가요?
- 자바스크립트는 개발자가 메모리 직접 접근하는 걸 허용하지 않음
- 허용되어 있다고 해도 값이 저장될 때 메모리 상황에 따라 임의로 주소가 결정되기 때문에 코드 실행 전에는 어디에 값이 저장되어 있는 지 알 수 없음
- 변수를 통해서 참조가 가능하기 때문에 변수 사용
